packages/hawtio/src/plugins/camel/routes/Source.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { jolokiaService } from '@hawtiosrc/plugins'
+import { CodeEditor, Language } from '@patternfly/react-code-editor'
+import { isRoutesFolder } from '@hawtiosrc/plugins/camel/camel-content-service'
+import { CamelContext } from '@hawtiosrc/plugins/camel/context'
+import { MBeanNode } from '@hawtiosrc/plugins/shared'
+
+export const findParentContext = (child: MBeanNode, root: MBeanNode): MBeanNode | undefined => {
+  let found: MBeanNode | undefined
+  root.children?.every(mb => {
+    if (mb.id === child.id) {
+      found = root
+      return false
+    } else found = findParentContext(child, mb)
+    return true
+  })
+
+  return found
+}
+export const Source: React.FunctionComponent = () => {
+  const { selectedNode, tree } = useContext(CamelContext)
+  const [xmlSource, setxmlSource] = useState<string>('')
+
+  useEffect(() => {
+    if (selectedNode) {
+      if (isRoutesFolder(selectedNode)) {
+        const contextMbean = findParentContext(selectedNode, tree.getTree()[0])
+
+        if (contextMbean?.objectName) {
+          jolokiaService.execute(contextMbean.objectName, 'dumpRoutesAsXml()').then(xml => {
```

Comment:
Since the `routes-service` has already performed this [function](https://github.com/hawtio/hawtio-next/blob/main/packages/hawtio/src/plugins/camel/routes-service.ts#L104) when run from the `tree-processor`, might it be useful to simply cache the xml on the node and then get it rather than executing the same request a second time?

Reply:
yeah, that make sense.